include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -506,11 +506,11 @@ enum class __peer_prefix_algo
 template <typename _OffsetT, __peer_prefix_algo _Algo>
 struct __peer_prefix_helper;
 
+#if (__LIBSYCL_VERSION >= 50700)
```

Comment:
Doesn't the use of `__dpl_sycl::__atomic_ref` make this helper available to work for earlier libsycl versions?
When `__LIBSYCL_VERSION < 50500`, then this maps to `sycl::atomic`.  Is there something else in this helper which needs protecting?

If so, should it get defined as `_ONEDPL_SYCL*_PRESENT` in `sycl_defs.h`  instead of relying on the version number directly in this code?

Reply:
`sycl::group_barrier` is not defined in older LIBSYCL versions. There is a `__dpl_sycl::__group_barrier` alternative, but it does not necessarily have the same semantics and is hard-coded to a work-group barrier, whereas this code needs a sub-group barrier.